internal/repository/cats/delete_cats_repository.go

```diff
@@ -0,0 +1,18 @@
+package catsRepository
+
+import (
+	"cats-social/internal/entity"
+	"context"
+	"fmt"
+)
+
+func (cr catRepository) Delete(ctx context.Context, catID string) error {
+    query := fmt.Sprintf(`UPDATE %s SET deleted_at = now() WHERE id = :catID`, entity.Cat{}.TableName())
+
+    _, err := cr.db.NamedExecContext(ctx, query, map[string]interface{}{"catID": catID})
+    if err != nil {
+        return err
+    }
```

Comment:
Bisa pake transaction biar isolated, cmiiw cc: @malikilamalik 
```suggestion
    tx := cr.db.MustBegin()
    _, err := tx.NamedExecContext(ctx, query, map[string]interface{}{"catID": catID})
    if err != nil {
        tx.Rollback()
        return err
    }
    tx.Commit()
```